python/ribasim/ribasim/model.py

```diff
@@ -376,6 +448,68 @@ def reset_contextvar(self) -> "Model":
         context_file_loading.set({})
         return self
 
+    def max_node_id(self) -> int:
+        return self.network.node.df.index.max()
+
+    def max_allocation_network_id(self) -> int:
+        m = self.network.node.df.allocation_network_id.max()
+        if pd.isna(m):
+            m = 0
+        return m
+
+    def merge_model(
+        self,
+        model_added: "Model",
+        offset_node_id: int | None = None,
+        offset_allocation_network_id: int | None = None,
+        offset_spacial: tuple[float, float] = (0.0, 0.0),
+        inplace: bool = True,
+    ):
+        """
+        Merge copies of the nodes and edges of an added model into this model.
+        The added model is not modified, but the following modificadions are made to the added data:
+        - Node IDs are shifted by at least the maximum node ID of this model
+        - Allocation network IDs are shifted by at least the maximum allocation network ID of this model.
+        """
+        if inplace:
+            model = self
+        else:
+            model = deepcopy(self)
+
+        nodes_model = model.nodes()
+        nodes_added = model_added.nodes()
+        nodes_added["network"] = nodes_added["network"].translate_spacially(
+            offset_spacial, inplace=False
+        )
+        min_offset_node_id = model.max_node_id()
+        min_offset_allocation_network_id = model.max_allocation_network_id()
+
+        if offset_allocation_network_id is None:
+            offset_allocation_network_id = min_offset_allocation_network_id
+        else:
+            assert (
+                offset_allocation_network_id >= min_offset_allocation_network_id
+            ), f"The allocation network ID offset must be at least the maximum allocation network ID of the main model ({min_offset_allocation_network_id}) to avoid conflicts."
```

Comment:
If you always check this, you might as well set the default argument to 0 instead of None.